### PR TITLE
fix(kaktoos-assimilator): move file open into try

### PR DESCRIPTION
**Pull request by BoySanic**
Moved file open function call into the try/catch block below it for fault tolerance in the case that the assimilator is unable to open the file.

### CODE REVIEW

File: images/makeproject/project/bin/kaktoos_assimilator.py [hunk 1]
```diff
@@ -18,12 +18,11 @@ def assimilate_handler(self, wu, results, canonical_result):
...
-
+                input_str = ""
                 try:
-                        os.makedirs(path)
+                    with open(input_path) as input_file:
+                        input_str = input_file.read()
+                    os.makedirs(path)
                 except OSError:
                         pass
                 with open(os.path.join(path, filename), "a") as f:
```
Comment (hube12) [inline images/makeproject/project/bin/kaktoos_assimilator.py:28]: Also guard this but instead if fail report the error in logs_error

**Review by hube12 — CHANGES_REQUESTED** [review]
#28

**Review by hube12 — APPROVED** [review]

File: images/makeproject/project/bin/kaktoos_assimilator.py [hunk 2]
```diff
@@ -18,18 +40,17 @@ def assimilate_handler(self, wu, results, canonical_result):
...
+                input_str = ""
                 try:
-                        os.makedirs(path)
+                    with open(input_path) as input_file:
+                        input_str = input_file.read()
+                    os.makedirs(path)
+                    with open(os.path.join(path, filename), "a") as f:
+                        for match in re_result.finditer(input_str):
+                            f.write("{}\n".format(match.group(1)))
                 except OSError:
```
Comment (hube12) [inline images/makeproject/project/bin/kaktoos_assimilator.py:51]: except OSError as e:
   print(e.filename)

**Comment by BoySanic** [comment]
I made the changes we originally discussed, with splitting the try/except block into 3 separate blocks for each error point we could hit

**Review by hube12 — APPROVED** [review]

**Review by ChipWolf — APPROVED** [review]

**Review by vcokltfre — APPROVED** [review]

**Pull request merged**